docs/FlaskSQLAlchemy.md

```diff
@@ -0,0 +1,418 @@
+    ```
+  * An init_app() method is used which is a way of constructing an instance of the particular package, then letting it know about the Flask instance (so that configuration details can be copied). Mechanically, it's just like any other instance method.
+    ```
+    db.init_app(app)
+    ```
+  * The Flask App can be loaded with the configuration of choice 
+    ```
+    application = create_app(get_env_config())
+    ```
+
+  **Note:** 
+  * Here the create_app() calls the get_env_config() which can be found 
+    [here](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L138) which contains the FLASK_ENVIORNMENT_CONFIG as "prod", "test", "dev", "local", "stag"
+    ```
+    def get_env_config() -> str:
+        flask_config_name = os.getenv("FLASK_ENVIRONMENT_CONFIG", "dev")
+        if flask_config_name not in ["prod", "test", "dev", "local", "stag"]:
+           raise ValueError(
+               "The environment config value has to be within these values: prod, dev, test, local, stag."
+      )
+        return CONFIGURATION_MAPPER[flask_config_name] 
+    ```
+
+   * The get_env_config() returns the [CONFIGURATION_MAPPER](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L147) as shown
+     ```
+     CONFIGURATION_MAPPER = {
+         "dev": "config.DevelopmentConfig",
+         "prod": "config.ProductionConfig",
+         "stag": "config.StagingConfig",
+         "local": "config.LocalConfig",
+         "test": "config.TestingConfig",
+     }
+     ```
+
+### Configuration
+
+The following configuration values exist for Flask-SQLAlchemy. Flask-SQLAlchemy loads these values from your main Flask config which can be populated in various ways. Note that some of those cannot be modified after the engine was created so make sure to configure as early as possible and to not modify them at runtime.
+
+A list of configuration keys currently understood by the extension can be found [here](https://flask-sqlalchemy.palletsprojects.com/en/2.x/config/)
+
+* The FLASK_ENV_CONFIG used in the the mentorship-backend are:
+  * prod
+  * test
+  * dev
+  * local
+  * stag
+
+* Among the various [configuration keys](https://flask-sqlalchemy.palletsprojects.com/en/2.x/config/) understood by the extension we take SQLALCHEMY_DATABASE_URI into consideration.
+
+**Note:** SQLALCHEMY_DATABASE_URI: The database URI that should be used for the connection. 
+URI or Uniform Resource Identifier is a string of characters that unambiguously identifies a particular resource.
+
+ * The SQLALCHEMY_DATABASE_URI for various FLASK_ENV_CONFIG are:
+
+   * [prod](https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L98)
+     ```
+     class ProductionConfig(BaseConfig):
+         """Production configuration."""
+
+         SQLALCHEMY_DATABASE_URI = BaseConfig.build_db_uri()
+         MOCK_EMAIL = False
+     ```
```

Comment:
@ashwinijha6 @keshakaneria instead of writng it like this, wouldn't it be better if we paste link of those lines here
LIke: `https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L98-103`
https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L98-L103

Reply:
Also, do we need to write flask env config and then sqlalchemy config at different places. May be both are same. Although I am not sure about this

Reply:
That is good over there i suppose

Reply:
@devkapilbansal I guess I have mentioned the link from where the code is taken
Do u want me to mention from line 98 to **103** means the 103?
Also I have mentioned the code for better understanding because the issue stated where sqlalchemy has been used in mentorship backend

Reply:
@devkapilbansal could u pls state the exact line where I have used flask env config & sqlalchemy config both

Reply:
I mean that instead of writing the config part by yourself give its link

Reply:
@devkapilbansal  I have given the link

Reply:
Leave it. I tried it manually but seems it work in comments only, not in markdown file :confused:

Reply:
yes https://github.com/anitab-org/mentorship-backend/blob/eae562267b1cbbb597ea3a8829b87ad373fa323c/config.py#L98-103
just reflects to the line 98 not the entire code block 98-103